src/main/java/com/rocketchat/core/APIServer.java

```diff
@@ -0,0 +1,22 @@
+package com.rocketchat.core;
+
+import com.rocketchat.models.chat.Chat;
+import com.rocketchat.models.user.User;
+import com.rocketchat.storage.InMemoryStorage;
+import com.rocketchat.storage.Storage;
+import com.rocketchat.websocket.core.BigQueue;
+import com.rocketchat.websocket.producers.MessageProducer;
+import com.rocketchat.websocket.producers.Producer;
+
+public class APIServer implements ServerBuilder {
+
+    @Override
+    public void build() {
```

Comment:
Si querés que sea heredable esta clase, quizás quieras agregar esto al método:

```suggestion
    public final void build() {
```

Esto te garantiza que el método no puede ser sobreescrito, aunque sí llamado por sus hijos. Esto ayuda al llamado Open/Closed Principle